platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/GuestClaimsHandlerExt.java

```
@@ -13,129 +13,208 @@
-            propertyMap = PropertiesLoader.toMap(properties);
+    /**
+     * Submit the name of the profile listed in each configuration to select that configuration
+     * for use. Applies retroactively for system-high and banner markings.
+     *
+     * @param selectedClaimsProfileName the name of the profile to be used.
+     */
+    public void setSelectedClaimsProfileName(String selectedClaimsProfileName) {
+        Map<String, Object> claimsProfiles = extractProfileNamePreserveResult(claimsProfileMaps);
+        if (!claimsProfiles.containsKey(selectedClaimsProfileName)
+                && !selectedClaimsProfileName.equals(DEFAULT_NAME)) {
+            LOGGER.warn(INVALID_GUEST_CLAIMS_MESSAGE);
+            throw new IllegalArgumentException(INVALID_GUEST_CLAIMS_MESSAGE);
         }
-        return propertyMap;
-    }
-
-    public void setAvailableClaimsFile(String availableClaimsFile) {
-        this.availableClaimsFile = availableClaimsFile;
+        this.selectedClaimsProfileName = selectedClaimsProfileName;
     }
 
-    public void setImmutableClaims(String claims) {
-        if (!StringUtils.isEmpty(claims)) {
-            immutableClaims.addAll(Arrays.asList(claims.split(",")));
+    /**
+     * Returns a map of claims configurations for the selected profile, or {@code null} if no
+     * profile has been selected.
+     *
+     * @return a map of claims configurations and attributes for writing to {@code users.attributes}.
+     */
+    @Nullable
+    public Map<String, Object> getSelectedClaimsProfileAttributes() {
+        Map<String, Object> claimsProfiles = extractProfileNamePreserveResult(claimsProfileMaps);
+        if (selectedClaimsProfileName != null && !selectedClaimsProfileName.equals(DEFAULT_NAME)) {
+            Object attributeListObject = claimsProfiles.get(selectedClaimsProfileName);
+            return ((Map<String, Object>) attributeListObject);
```

Yes but if it does we have bigger problems and we should let the exception be thrown. It will be caught higher up as a runtime exception and we will log the failure. We _need_ that attribute list object to be the specified type: Map<String, Object> or we will break config admin.